### PR TITLE
Add commission rate

### PR DESCRIPTION
**Pull request by stevenlcf**

### CODE REVIEW

File: contracts/DPoS.sol [hunk 1]
```diff
@@ -121,14 +128,45 @@ contract DPoS is IDPoS, Ownable {
         registeredSidechains[_addr] = true;
     }
 
-    function initializeCandidate(uint _minSelfStake) external {
+    function initializeCandidate(uint _minSelfStake, uint _commissionRate, uint _rateLockEndTime) external {
         ValidatorCandidate storage candidate = candidateProfiles[msg.sender];
         require(!candidate.initialized, "Candidate is initialized");
+        require(_commissionRate <= 10000);
```
Comment (Xuefeng-Zhu) [inline contracts/DPoS.sol:134]: avoid hard coding

File: contracts/DPoS.sol [hunk 2]
```diff
@@ -121,14 +128,45 @@ contract DPoS is IDPoS, Ownable {
...
+        candidate.announcedLockEndTime = _newLockEndTime;
+        candidate.announcementTime = block.number;
+
+        emit CommissionRateAnnouncement(msg.sender, _newRate, _newLockEndTime);
+    }
+
+    function confirmIncreaseCommissionRate() external {
+        ValidatorCandidate storage candidate = candidateProfiles[msg.sender];
+        require(candidate.initialized, "Candidate is not initialized");
+        require(block.number > candidate.announcementTime + 1344, "new rate hasn't taken effect"); // TODO: use a const (1344 blocks = 2 weeks)
```
Comment (Xuefeng-Zhu) [inline contracts/DPoS.sol:167]: avoid hard coding

File: contracts/DPoS.sol (hunk 1 above)
Comment (stevenlcf) [reply contracts/DPoS.sol:134]: As I just mentioned during the meeting, this parameter will be updated to a configurable param in governance PR. I think we can leave it as it is in this PR.

File: contracts/DPoS.sol (hunk 2 above)
Comment (stevenlcf) [reply contracts/DPoS.sol:167]: Same as the last comment.

File: contracts/DPoS.sol (hunk 1 above)
Comment (xiaozhou) [reply contracts/DPoS.sol:134]: Should not put the number directly here, but I think this value can be a constant, no need for governance.

**Pull request merged**